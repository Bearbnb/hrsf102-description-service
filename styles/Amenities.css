/* .modalBackdrop {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  height: 1500px;
  overflow-y: auto;
  z-index: 1041;
  background: rgba(255, 255, 255, 0.3);
}

.modalMain {
  position: fixed;
  background: white;
  margin-top: 70px;
  min-width: 450px;
  height: auto;
  overflow-y: auto; 
  top: 50%;
  left: 50%;
  transform: translate(-50%,-50%);
  box-shadow: 0 0 7px #C8C8C8;
} */

.modalBackdrop {
  position: fixed;
  background: rgba(255, 255, 255, 0.6);
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: 1;
  overflow-y: scroll;
}

.modalMain {
  box-shadow: 0 0 7px #C8C8C8;
  background: white;
  max-width: 450px;
  margin: 80px auto 0px auto;
  height: auto;
  z-index: 2;
  padding: 20px;
}

.displayBlock {
  display: block;
}

.displayNone {
  display: none;
}

.title {
  color: #484848;
  margin-left: 23px;
}