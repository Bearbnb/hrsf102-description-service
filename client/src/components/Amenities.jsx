import React from 'react';
import PropTypes from 'prop-types';

import Amenity from './Amenity.jsx';

import { generateAmenitiesArray } from '../../../seedDatabase/helpers';
import styles from '../../../styles/Amenities.css';

const Amenities = ({
  amenities,
  hideAmenities,
  showModal,
  hideAmenitiesPress,
}) => {
  const display = showModal ? styles.displayBlock : styles.displayNone;

  const amenitiesArray = generateAmenitiesArray(amenities);
  return (
    <div
      className={`${display} ${styles.modalBackdrop}`}
      onClick={() => hideAmenities()}
    >
      <div 
        className={`${styles.modalMain}`}
      >
        <h1 className={styles.title}>Amenities</h1>
        {amenitiesArray.map((amenity, i) => <Amenity key={i} availableAmenity={amenity} />)}
      </div>

    </div>
  );
};

Amenities.propTypes = {
  amenities: PropTypes.shape({
    id: PropTypes.number,
    _id: PropTypes.string,
    basics: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    bedAndBath: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    dining: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    guestAccess: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    safetyFeautes: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    notIncluded: PropTypes.arrayOf(PropTypes.string),
  }),
  hideAmenities: PropTypes.func,
  hideAmenitiesPress: PropTypes.func,
  showModal: PropTypes.bool,
};

Amenities.defaultProps = {
  amenities: {},
  hideAmenities: () => {},
  hideAmenitiesPress: () => {},
  showModal: false,
};

export default Amenities;
