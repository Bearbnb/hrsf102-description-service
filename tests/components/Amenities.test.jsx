import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Amenities from '../../client/src/components/Amenities';

describe('<Amenities />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Amenities amenities={{}} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
