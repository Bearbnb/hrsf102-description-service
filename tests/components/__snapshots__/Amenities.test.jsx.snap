// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Amenities /> should render properly 1`] = `
<div
  className="displayNone modalBackdrop"
  onClick={[Function]}
>
  <div
    className="modalMain"
  >
    <h1
      className="title"
    >
      Amenities
    </h1>
  </div>
</div>
`;
